Issue 9: Submit with field display values as fieldname

Jesse-Springston opened the pull request:
- Changed submit_request to accept inputs with display values as field IDs (ex: {"id": "How are you?", "value": "Good!"} instead of {"id": "mood", "value": "Good!"}

- Fixed regular expression error by changing re patterns to raw strings

DanielJDufour reviewed (approved):
Awesome!

DanielJDufour commented on gogov/__init__.py:
```diff
@@ -383,14 +384,23 @@ def submit_request(
         # Make a single dict with {id}: {value} for each field dict in "fields" for input validation
         input_fields = {field["id"]: field["value"] for field in fields}
```
Let's get this committed after addressing the question about ["5825"], but I'm thinking for the next phase, it might be a nicer UX if someone could just pass in the fields as a simple dictionary vs. having to pass in an array of dictionary like:
```
submit_request({
    fields={
        "How many Potholes?": 4,
        "Work Recently Done?": "no"
    }
})
```

Jesse-Springston replied:
Sure. Do you want me to preserve the current functionality, or just require the input be in this format?

Jesse-Springston commented on gogov/__init__.py:
```diff
@@ -383,14 +384,23 @@ def submit_request(
         input_fields = {field["id"]: field["value"] for field in fields}
 
+        # A dictionary of {fieldname}: {display value} for checking if the user input the display values in fields
+        custom_fields = self.get_custom_fields()["58525"]
```
I'm not sure why, but the get_custom_fields function returns the dictionary with the fieldnames and display values inside another dictionary with a lone key, "58525". I tried the function on multiple occasions to make sure that value wasn't changing. I suppose I could get around the specific key name by doing a for loop. If there's only one key in the outer dictionary, it shouldn't affect runtime.

Jesse-Springston commented:
New commit for this PR:

- Addressed fields input being too complex. The function no longer enforces GOGov format for the user, and accepts a simple dictionary of fieldname: value. 
- Also simplified inputting the location. The user is no longer required to input a location dictionary in the GOGov format; they simply need to provide an address and (optionally) values for longitude and latitude (these will be determined using geochatt on the address, if not provided by the user).

jhardin-ChattGov commented:
Changes look good. That dictionary key generation to check for inputs was interesting. User input looks like it'll be robust with this.
